tileserver/__init__.py

```diff
@@ -273,12 +298,19 @@ def handle_request(self, request):
                 tile_data = reformat_selected_layers(
                     tile_data_all, self.layer_config.all_layers, coord, format)
 
-                # note that we want to store the data too, as this means that
-                # future requests can be serviced directly from the store.
+                # note that we want to store the formatted data too,
+                # as this means that future requests can be serviced
+                # directly from the store.
                 if self.store and coord.zoom <= 20:
                     self.io_pool.apply_async(
                         async_store, (self.store, tile_data, coord, format))
 
+            # since we're a request for all, enqueue the coordinate to
+            # ensure other formats get processed
+            if self.sqs_queue and coord.zoom <= 20:
```

Comment:
Do we only want to do this when `layer_spec == 'all'`, or is it worth doing this always? I guess what I'm trying to figure out is: If we add something to the TOI, should we always enqueue work to render the tile as if the tile were originally seeded?

The way the code is at the moment, we always add the tile to the TOI (e.g: perhaps it's a `buildings` layer request), and we'd store the JSON and (if any) other format that was requested, but not _all_ the formats.